test/request.js

```diff
@@ -92,5 +107,47 @@ describe('request', () => {
         responseStub.emit('end');
     });
 
+    ['gzip', 'deflate'].forEach((encoding) => {
+        it(`should inflate response body with ${encoding} encoding`, (done) => {
```

Comment:
I feel that this test and the next one are too similar. Extracting out common code is probably a good idea. Or at least pulling out creation of a gzip stream, as this is quite unreadable in the test.